src/engine/strat_engine/thinpool.rs

```diff
@@ -166,14 +166,14 @@ impl ThinPool {
 
     /// The status of the thin pool as calculated by DM.
     pub fn check(&mut self, dm: &DM) -> EngineResult<ThinPoolStatus> {
-        let thinpool = try!(self.thin_pool.status(dm));
-        try!(self.mdv.check());
+        let thinpool = self.thin_pool.status(dm)?;
+        self.mdv.check()?;
 
-        let filesystems: Vec<_> = try!(self.filesystems
-                                           .borrow_mut()
-                                           .into_iter()
-                                           .map(|fs| fs.check(dm))
-                                           .collect());
+        let filesystems = self.filesystems
```

Comment:
That, and mostly the preferred way is to put it after collect. It allows collect to do a bit of magic here and there.